src/Data/MonadicStreamFunction/Util.hs

```diff
@@ -67,6 +67,17 @@ mapMSF = MSF . consume
       (bs, sf'') <- consume sf' as
       b `seq` return (b:bs, sf'')
 
+-- | Apply an 'MSF' to every input, discarding results.
+-- Used for side-effects only.
+mapMSF_ :: Monad m => MSF m a b -> MSink m [a]
```

Comment:
That's maybe useful, but can't you rewrite it in terms of `mapMSF`? Something like `mapMSF . (>>> arr void)`. (I don't think performance will suffer, because laziness.) Using the `MSF` constructor is not encouraged.